### DISCUSSION TITLE
[useRangeCalendar] Whether behavior of "Stop range selection" is a right design?

### DISCUSSION THREAD

**Discussion by QzCurious** [adobe/react-spectrum#5703] [Q&A] [answered]
# How I think it's not a right design

As a user, I would not expect that clicking on title of calendar, empty cell or anywhere outside the calendar would affect my selection. But:

https://github.com/adobe/react-spectrum/assets/38932402/8637eb7b-dffa-43d4-a6a0-e466d396be51

_The running code is an example from [docs](https://react-spectrum.adobe.com/react-aria/useRangeCalendar.html#styled-examples)_

Digging into source code I found the code doing that

https://github.com/adobe/react-spectrum/blob/4ae2831f9487958511c88b9ebe672b5650042d7e/packages/%40react-aria/calendar/src/useRangeCalendar.ts#L66-L71

The problem is that `e.relatedTarget` would be `body` if I clicked a un-focusable element. So `!ref.current.contains(e.relatedTarget)` would be `true` and so my selection will be fulfilled by the code.

# Real world example for interacting on title, empty cell, etc.

I want to make a scrollable RangeCalendar as Booking.com:

https://github.com/adobe/react-spectrum/assets/38932402/3531b5c3-0e04-4f28-9d9c-ad0d6ff2dcb7

Here's a [codesandbox](https://codesandbox.io/p/sandbox/react-aria-scrollable-range-calendar-h78c39) with `useRangeCalendar` (modified from the code [example](https://react-spectrum.adobe.com/react-aria/useRangeCalendar.html#styled-examples)):

https://github.com/adobe/react-spectrum/assets/38932402/7a620bd2-145d-4e02-95be-cc379e499254

But when I scroll RangeCalendar by "finger" it select the same day as end day. I can hardly select range across multiple months "on mobile". It is caused by `endDragging`, which has similar behavior to `onBlur` aforementioned.

https://github.com/adobe/react-spectrum/blob/4ae2831f9487958511c88b9ebe672b5650042d7e/packages/%40react-aria/calendar/src/useRangeCalendar.ts#L41-L61

# ~A possible work around~

> I've add some comment on my [code](https://codesandbox.io/p/sandbox/objective-shape-8r4utm?file=%2Fsrc%2FApp.js%3A10%2C1-15%2C9) that overwrite `onBlur` to noop and re-implement to behavior by detecting outside click.
It's just a simple work around. Not considering tabbing, the `onPress` logic, etc.

I was wrong about the work around. It only fix `onBlur` part, which is possible to be overwrited.

**Comment by LFDanLu** [comment]
I think https://github.com/adobe/react-spectrum/issues/3738#issuecomment-1741041887 had an interesting idea for handling this where we should ignore blur events that happen as a result of an interaction within the RangeCalendar's containing modal. Of course there is still the issue where you could be pressing on a non-focusable element like you've shown here so maybe we could try to improve detection here or disable this blur behavior as a whole when in a modal and instead tie it to the modal's closure instead.

**Comment by QzCurious** [comment, marked as answer]
I'm closing this as my goal is to make a mobile-scrollable-multi-month calendar. But I made it too vague. Virtually, touch-scrolling a calendar is not necessary related to clicking, tabbing out a calendar.

A new issue with PR for on touch-scroll only: #5975
